airbyte-cdk/bulk/toolkits/load-s3/src/main/kotlin/io/airbyte/cdk/load/file/s3/S3Client.kt

```diff
@@ -44,6 +47,8 @@ class S3Client(
     val bucketConfig: S3BucketConfiguration,
     private val uploadConfig: ObjectStorageUploadConfiguration?,
 ) : ObjectStorageClient<S3Object> {
+    private val log = KotlinLogging.logger {}
+    private val uploadPermits = uploadConfig?.maxNumConcurrentUploads?.let { Semaphore(it) }
```

Comment:
I was just thinking about what the max number of concurrent number of uploads we allow in general. Not blocking this PR, but do we want to eventually into the CDK directly?

Reply:
My plan was to leave this in as part of the internal configuration, with a reasonable default